flink-libraries/flink-table/src/main/scala/org/apache/flink/table/plan/rules/dataSet/DataSetAggregateRule.scala

```diff
@@ -53,8 +54,17 @@ class DataSetAggregateRule
 
     // check if we have distinct aggregates
     val distinctAggs = agg.getAggCallList.exists(_.isDistinct)
+    if (distinctAggs) {
+      throw TableException("DISTINCT aggregates are currently not supported.")
```

Comment:
We should not throw an exception here, but return with `false` from the `matches()` call.
Otherwise the optimization terminates, although the query could be rewritten without distinct aggregates.